textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -120,14 +132,49 @@ public void open() throws Exception, DataFlowException {
      */
     @Override
     public ITuple getNextTuple() throws Exception {
+    	if (cursor == CLOSED) {
+            throw new DataFlowException(ErrorMessages.OPERATOR_NOT_OPENED);
+        }
+
         if (innerTupleList.isEmpty()) {
             return null;
         }
 
+        if (resultCursor >= limit + offset - 1 || limit == 0){
+            return null;
+        }
+
         ITuple nextTuple = null;
 
+        if (offset > 0) {
```

Comment:
In my opinion, it would be better if you could put this part of logic into a function called "computeNextMatchingTuple" to be consistent with other operators. You can take a look at the function "computeNextMatchingTuple" in keywordMatcher, they should have similar structures.